src/dg/dg.cpp

```diff
@@ -2124,46 +2278,158 @@ void DGBase<dim,real,MeshType>::evaluate_mass_matrices (bool do_inverse_mass_mat
+        const unsigned int n_metric_dofs = high_order_grid->fe_system.dofs_per_cell;
+        std::vector<dealii::types::global_dof_index> metric_dof_indices(n_metric_dofs);
+        metric_cell->get_dof_indices (metric_dof_indices);
+        const unsigned int grid_degree = high_order_grid->fe_system.tensor_degree();
+        //get mapping_support points
+        std::vector<std::vector<real>> mapping_support_points(dim);
+        for(int idim=0; idim<dim; idim++){
+            mapping_support_points[idim].resize(n_metric_dofs/dim);
+        }
+        dealii::QGaussLobatto<dim> vol_GLL(grid_degree +1);
+        for (unsigned int igrid_node = 0; igrid_node< n_metric_dofs/dim; ++igrid_node) {
+            for (unsigned int idof = 0; idof< n_metric_dofs; ++idof) {
+                const real val = (high_order_grid->volume_nodes[metric_dof_indices[idof]]);
+                const unsigned int istate = fe_metric.system_to_component_index(idof).first; 
+                mapping_support_points[istate][igrid_node] += val * fe_metric.shape_value_component(idof,vol_GLL.point(igrid_node),istate); 
+            }
+        }
 
-        for (unsigned int itest=0; itest<n_dofs_cell; ++itest) {
+        //get determinant of Jacobian
+        std::vector<real> determinant_Jacobian(n_quad_pts);
+        operators.build_local_vol_determinant_Jac(grid_degree, fe_index_curr_cell, n_quad_pts, n_metric_dofs/dim, mapping_support_points, determinant_Jacobian);
 
-            const unsigned int istate_test = fe_values_volume.get_fe().system_to_component_index(itest).first;
+ 
+        if(this->all_parameters->use_weight_adjusted_mass == false){
+           // const std::vector<real> &JxW = fe_values_volume.get_JxW_values();
+            std::vector<real> JxW(n_quad_pts);
+         //   const std::vector<real> &JxW_dealii = fe_values_volume.get_JxW_values();
+            for(unsigned int iquad=0; iquad<n_quad_pts; iquad++){
+                JxW[iquad] = quad_weights[iquad] * determinant_Jacobian[iquad];
+            }
+            operators.build_local_Mass_Matrix(JxW, n_dofs_cell, n_quad_pts, fe_index_curr_cell, local_mass_matrix);
 
-            for (unsigned int itrial=itest; itrial<n_dofs_cell; ++itrial) {
+            std::vector<dealii::FullMatrix<real>> local_mass_matrix_aux(dim);
+            for(int idim=0; idim<dim; idim++){
+                local_mass_matrix_aux[idim].reinit(n_dofs_cell, n_dofs_cell);
+                if(use_auxiliary_eq){
+                    local_mass_matrix_aux[idim].add(1.0,local_mass_matrix);
+                }
+            }
 
-                const unsigned int istate_trial = fe_values_volume.get_fe().system_to_component_index(itrial).first;
+            if(this->all_parameters->flux_reconstruction_type != Parameters::AllParameters::Flux_Reconstruction::cDG){
+                //For flux reconstruction
+                dealii::FullMatrix<real> K_operator(n_dofs_cell);
```

Comment:
K_operator means nothing. Please give it relevant name, or specific equation line of a reference.

Reply:
fixed